Lib/asyncio/base_events.py

```diff
@@ -706,6 +706,8 @@ def call_later(self, delay, callback, *args, context=None):
         Any positional arguments after the callback will be passed to
         the callback when it is called.
         """
+        if delay is None:
+            raise TypeError('delay must not be None')
```

Comment:
It is redundant. A TypeError will be raised at the next line, by `self.time() + delay`.

Reply:
My thought when I suggested the change was: better to explicitly warn about `None` argument than a little cryptic `unsupported operand type(s) for +: 'NoneType' and 'int'`.
I believe not only exception type but also a proper message can help users to avoid silly mistakes.

Reply:
Why is `None` special for this argument? I do not know any precedents of checkinjg only for None if other types can be invalid too. It looks like an antipattern to me. We either check for allowed types, or do not check either and let an error be raised later when perform an unsupported operation.